### ISSUE TITLE
Return type "unsigned int constexpr" not parsed properly

### ISSUE THREAD

**Issue by ralfulrich** [doxygen/doxygen#8933]
**Describe the bug**

In a valid and working C++17 header file, whenever I use return types of type `unsigned int constexpr` this results in sphinx WARNINGs since apparently it mixes up while parsing my code (see output below). For example, the perfectly fine C++17 function:
```
/**
   * Get the nucleus mass number A.
   *
   * @param code for nucleus
   * @return unsigned int constexpr
   */
  inline unsigned int constexpr get_nucleus_A(Code const code) {
    return (static_cast<CodeIntType>(code) % 10000) / 10;
  }
```
produces the following sphinx WARNING:
```
Running Sphinx v3.5.3

 WARNING: Error when parsing function declaration.
If the function has no return type:                                                                                                                                                                                                                             
  Error in declarator or parameters-and-qualifiers                                                                                                                                                                                                              
  Invalid C++ declaration: Expected identifier in nested name, got keyword: unsigned [error at 8]                                                                                                                                                               
    unsigned constexpr int get_nucleus_A (Code const)                                                                                                                                                                                                           
    --------^                                      
```

If I replace `unsigned int` with `size_t` this warning disappears, but isn't there another less intrusive way to fix this? 


**Expected behavior**
No warning, and no "unsigned constexpr int"

**Version**
This is on ubuntu 20.04 using doxygen 1.8.17 

Originally asked at: https://stackoverflow.com/questions/69882899/unsigned-int-constexpr-in-doxygen-sphinx-c17

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8939

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
